server/api/image.py

```diff
@@ -94,6 +98,84 @@ def find(self, params):
         ]
 
     @describeRoute(
+        Description('Download multiple images as a ZIP file.')
+        .param('datasetId', 'The ID of the dataset to download.',
+               required=False)
+        .param('filter', 'Filter the images by a PegJS-specified grammar '
+                         '(causing "datasetId" to be ignored).',
+               required=False)
+        .errorResponse()
+    )
+    @access.public
+    def downloadMultiple(self, params):
+        Dataset = self.model('dataset', 'isic_archive')
+        File = self.model('file')
+        Image = self.model('image', 'isic_archive')
+
+        if 'filter' in params:
+            query = querylang.astToMongo(json.loads(params['filter']))
```

Comment:
Now if I browse to Lesion Images, don't filter anything, and click "Download as ZIP" I see a JSON error. This happens when `getFilterAstTree()` returns `undefined`, which is added literally to the URL: `http://localhost:8080/api/v1/image/download?filter=undefined`.